common_logging/src/main/java/uk/gov/dhsc/htbhf/logging/Event.java

```diff
@@ -0,0 +1,22 @@
+package uk.gov.dhsc.htbhf.logging;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import lombok.Builder;
+import lombok.Data;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+
+@Data
+@Builder
+public class Event {
+    private EventType eventType;
+    private LocalDateTime timestamp;
+    private Map<String, Object> eventMetadata;
+
+    //This is here to make sure that the field "eventMetadata" isn't shown in the logs
+    @JsonAnyGetter
+    public Map<String, Object> getEventMetadata() {
+        return eventMetadata;
```

Comment:
Does the use of a map here mean that the order of fields logged as json will be indeterminate? If so, can we `return new TreeMap<String, Object>(eventMetadata)` instead?

Reply:
what happens if we provide a null map?

Reply:
If we provide a null Map then no metadata is shown in the JSON at all, it is handled nicely by default.